developer/src/kmc-kmn/src/ng-compiler/token-buffer.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ *
+ * Created by Dr Mark C. Sinclair on 2025-03-13
+ *
+ * KMC KMN Next Generation Parser (TokenBuffer)
+ */
+
+import { Token, TokenTypes } from "./lexer.js";
+
+export class TokenBuffer {
+  private list: Token[];
+  private _cp: number;
+  private eof: Token = new Token(TokenTypes.TT_EOF, "");
+
+  public constructor(list: Token[]) {
+    this.list = list;
+    this._cp = 0;
+  }
+
+  public get currentPosition() { return this._cp; }
+
+  public nextToken(): Token {
+    return this._cp == this.list.length ? this.eof : this.list[this._cp];
+  }
+
+  public popToken(): void {
+    this._cp = this._cp + 1 <= this.list.length ? this._cp + 1 : this._cp;
```

Comment:
```suggestion
    this._cp = Math.max(this._cp + 1, this.list.length);
```

if my logic is correct?

Reply:
Sadly, logic wasn't correct:
`this._currentPosition = Math.min(this._currentPosition + 1, this.list.length);`

Reply:
0 for 2!